EIPS/eip-5851.md

```diff
@@ -0,0 +1,240 @@
+---
+eip: 5851
+title: ZK-based-KYC-certificates
+description: Interface for assigning/validating identities using Zero Knowledge Proofs
+author: Yu Liu (@yuliu-debond), Dhruv Malik (@dhruvmalik007)
+discussions-to: https://ethereum-magicians.org/t/eip-5185-kyc-certification-issuer-and-verifier-standard/11513
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-10-18
+requires: 721, 1155, 3643, 5114
+---
+
+
+## Abstract
+
+- This EIP Provides defined interface for KYC verification with abstract onchain conditions.
+
+- This EIP defines the necessary interface for orchestrator to assign identity certificates (as Soulbound tokens) to the wallets, which can be verified by ZK schemes.
+
+## Motivation
+
+Onchain verification is becoming indispensable across DeFI as well as other web3 protocols (DAO, governance) as its needed not only by the government for regulatory purposes, but also by different DeFI protocols to whitelist the users which fullfill the certain criterias.This created the necessity of building onchain verification of the addresses for token transfers (like stablecoin providers check for the blacklisted entities for the destination address, limited utility tokens for a DAO community , etc). Along with the concern that current whitelisting process of the proposals are based on the addition of the whitelisted addresses (via onchain/offchain signatures) and thus its not trustless for truly decentralised protocols.
+
+Also Current standards in the space, like [eip-3643](./eip-3643.md) are insufficient to handle the complex usecases where:
+
+- The validation logic needs to be more complex than verification of the user identity wrt the blacklisted address that is defined offchain, and is very gas inefficient.
+
+- Also privacy enhanced/anonymous verification is important need by the crypto users in order to insure censorship/trustless networks. ZK based verification schemes are currently the only way to validate the assertion of the identity by the user, while keeping certain aspects of the providers identity completely private.
+
+thus in order to address the above major challenges: there is need of standard that defines the interface of contract which can issue an immutable identity for the identifier (except by the user) along with verifying the identity of the user based on the ownership of the given identity token.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Definitions
```

Comment:
Suggestion: define ZK, as it might not be a well-known acronym

```suggestion
### Definitions

- ZK: Zero-knowledge. Refers to a cryptographic system in which a characteristic of an entity can be proven without revealing which entity the proof refers to.
```